src/schema.py

```diff
@@ -95,15 +99,104 @@ class Meta:
         model = CapacityModel
 
 
+# MARK: - User
+
+
+class User(SQLAlchemyObjectType):
+    class Meta:
+        model = UserModel
+
+
+class UserInput(graphene.InputObjectType):
+    net_id = graphene.String(required=True)
+    giveaway_id = graphene.Int(required=True)
+
+
+# MARK: - Giveaway
+
+
+class Giveaway(SQLAlchemyObjectType):
+    class Meta:
+        model = GiveawayModel
+
+    user_ids = graphene.List(lambda: User)
+
+    def resolve_user_ids(self, info):
+        query = User.get_query(info=info).filter(UserModel.giveaway_id == self.id)
+        return query
+
+
+# MARK: - Giveaway
+
+
+class GiveawayInstance(SQLAlchemyObjectType):
+    class Meta:
+        model = GiveawayInstanceModel
+
+
 # MARK: - Query
 
 
 class Query(graphene.ObjectType):
     gyms = graphene.List(Gym)
+    users_by_giveawayid = graphene.List(User, id=graphene.Int())
 
     def resolve_gyms(self, info):
         query = Gym.get_query(info)
         return query.all()
 
 
-schema = graphene.Schema(query=Query)
+def resolve_users_by_giveaway_id(self, info, id):
+    query = User.get_query(info).filter(UserModel.giveaway_ids.any(GiveawayInstance.giveaway_id == id))
+    return query.all()
+
+
+# MARK: - Mutation
+
+
+class CreateUser(graphene.Mutation):
+    class Arguments:
+        net_id = graphene.String()
+
+    user = graphene.Field(User)
+
+    def mutate(root, info, net_id):
+        existing_user = db_session.query(UserModel).filter(UserModel.net_id == net_id).first()
+        if existing_user:
+            return CreateUser(user=existing_user)
+
+        new_user = UserModel(net_id=net_id, giveaway_ids=[])
+        db_session.add(new_user)
+        db_session.commit()
+        return CreateUser(user=new_user)
```

Comment:
```suggestion
    def mutate(root, info, net_id):
        # Check to see if NetID already exists
        existing_user = User.get_query(info).filter(UserModel.net_id == net_id).first()
        if existing_user:
            raise GraphQLError("NetID already exists.")

        # NetID does not exist
        new_user = UserModel(net_id=net_id)
        db_session.add(new_user)
        db_session.commit()
        return CreateUser(user=new_user)
```

Let's keep the querying consistent and avoid using the query function from the DB session. You also want to add proper error handling so that frontend can process the errors. You can simply raise a `GraphQLError` and pass in a string with the error message. Make sure to import it using `from graphql import GraphQLError`.